sagemaker_huggingface_inference_toolkit/handler_service.py

```diff
@@ -0,0 +1,213 @@
+# Copyright 2021 The HuggingFace Team, Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import importlib
+import logging
+import os
+import time
+import sys
+from abc import ABC
+
+from sagemaker_inference import environment
+from transformers.pipelines import SUPPORTED_TASKS
+
+from mms.service import PredictionException
+from mms import metrics
+from sagemaker_huggingface_inference_toolkit import decoder_encoder
+from sagemaker_huggingface_inference_toolkit.transformers_utils import (
+    _is_gpu_available,
+    get_pipeline,
+    infer_task_from_model_architecture,
+)
+
+
+ENABLE_MULTI_MODEL = os.getenv("SAGEMAKER_MULTI_MODEL", "false") == "true"
```

Comment:
We should call out that downloading HF Model on the fly will not be supported for multi-model endpoint.